src/main/cpp/auton/drivePrimitives/DrivePathPlanner.cpp

```diff
@@ -228,22 +227,24 @@ bool DrivePathPlanner::IsDone()
         return true;
     }
 
-    if (m_isVisionDrive)
+    if (m_isVisionDrive && m_driveToObject != nullptr)
     {
         Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "DrivePathPlanner", m_driveToObject->GetDriveStateName(), m_driveToObject->IsDone());
         return m_driveToObject->IsDone();
     }
-
-    if (m_checkForDriveToUpdate && !m_isVisionDrive)
+    else if (!m_isVisionDrive)
     {
-        CheckForDriveTo();
+        auto trajDrivePathPlanner = dynamic_cast<TrajectoryDrivePathPlanner *>(m_chassis->GetSpecifiedDriveState(ChassisOptionEnums::TRAJECTORY_DRIVE_PLANNER));
+        if (trajDrivePathPlanner != nullptr && m_driveToObject == nullptr)
```

Comment:
I think this is fine, but I think we will never hit the trajectoryDrivePathPlanner is done if we have a zone/drive to object